shunting-yard.h

```diff
@@ -88,7 +90,19 @@ struct TokenUnary : public TokenBase {
 };
 
 class packToken;
-typedef std::queue<TokenBase*> TokenQueue_t;
+
+// Adapt to std::queue<TokenBase*>
+class TokenQueue_t: public std::deque<TokenBase*> {
+public:
+  void push(TokenBase* t) {
+    push_back(t);
+  }
+  void pop() {
+    pop_front();
+  }
+};
+
+
```

Comment:
I don't think this change is necessary, do you have an argument in favor of keeping this? Since this is not even being used I am not sure we should keep it.

Reply:
Oh I haven't noticed the type changed xP, that's why I did this comment. My fault here.